### PR TITLE
Creating first version of bamToFastq

### PR DESCRIPTION
**Pull request by MurtoHilali**

### CODE REVIEW

File: bamToFastq.wdl [hunk 1]
```diff
@@ -0,0 +1,341 @@
...
+    }
+
+    if (nameCheck.valid == true){
+        call backExtract { 
+            input:
+                bamFile = bamFile
+        }
+    }
+
+    call renameFastqs {
```
Comment (pruzanov) [inline bamToFastq.wdl:61]: Maybe move this under #54 if (nameCheck.valid == true) ? the task backExtract is called only if nameCheck is successfull

File: bamToFastq.wdl [hunk 2]
```diff
@@ -0,0 +1,341 @@
...
+            modules: "Required environment modules"
+            memory: "Memory allocated for this job"
+            timeout: "Time in hours before task timeout"
+        }
+
+        command <<<
+
+            module load samtools
+            samtools flagstat ~{bamFile} > ~{prefix}.flagstat.txt
+            samtools view -H ~{bamFile} | grep -G "^@RG" > readgroups.tsv
```
Comment (pruzanov) [inline bamToFastq.wdl:96]: according to grep documentation -G (basic grep) option is enabled by default

File: bamToFastq.wdl [hunk 3]
```diff
@@ -0,0 +1,341 @@
...
+            module unload cromwell ## <- only needed for local testing
+            module unload java ## <- only needed for local testing
+            module load picard
+
+            java -Xmx20g -jar $PICARD_ROOT/picard.jar SamToFastq \
+            INPUT=~{bamFile} \
+            RG_TAG="ID" \
+            OUTPUT_DIR=. \
+            OUTPUT_PER_RG=true \
+            NON_PF=true \
```
Comment (pruzanov) [inline bamToFastq.wdl:258]: I would consider exposing at least some of these variables, perhaps NON_PF and OUTPUT_PER_RG? You may always set them to pre-selected default values

**Review by pruzanov — APPROVED** [review]

File: bamToFastq.wdl [hunk 4]
```diff
@@ -0,0 +1,343 @@
...
+
+        command <<<
+            module unload cromwell ## <- only needed for local testing
+            module unload java ## <- only needed for local testing
+            module load picard
+
+            java -Xmx20g -jar $PICARD_ROOT/picard.jar SamToFastq \
+            INPUT=~{bamFile} \
+            RG_TAG="ID" \
+            OUTPUT_DIR=. \
```
Comment (xuemeiluo) [inline bamToFastq.wdl:256]: OUTPUT_DIR may not be needed

**Review by lheisler — APPROVED** [review]

**Pull request merged**